include/Model/Languages.php

```diff
@@ -747,6 +747,48 @@ public function delete_transient_from_options_table(): void {
 		}
 	}
 
+
+	/**
+	 * Returns the language arguments from the locale and additional arguments.
+	 *
+	 * @since 3.8
+	 *
+	 * @param array $args Arguments to create a language.
+	 * @return array|false Array of language arguments or false if the locale is invalid.
+	 *
+	 * @phpstan-param array{locale: string, name?: string, slug?: string, is_rtl?: bool, flag_code?: string, term_group?: int, no_default_cat?: bool} $args
+	 * @phpstan-return array{locale: string, name: string, slug: string, rtl: bool, flag: string, term_group: int|numeric-string, no_default_cat: bool}|false
+	 */
+	public function get_language_args_from_locale( array $args ) {
+		if ( isset( $args['name'], $args['slug'], $args['is_rtl'], $args['flag_code'] ) ) {
+			return array(
+				'locale'         => $args['locale'],
+				'name'           => $args['name'],
+				'slug'           => $args['slug'],
+				'rtl'            => $args['is_rtl'],
+				'flag'           => $args['flag_code'],
+				'term_group'     => $args['term_group'] ?? 0,
+				'no_default_cat' => $args['no_default_cat'] ?? false,
+			);
+		}
+
+		$languages = include POLYLANG_DIR . '/settings/languages.php';
+
+		if ( empty( $languages[ $args['locale'] ] ) ) {
+			return false;
+		}
```

Comment:
If I correctly understand the @chouby [comment](https://github.com/polylang/polylang-pro/pull/2653#discussion_r2251762193), the goal is to allow an unknown locale. But it seems to me that isn't the case here, isn't it?

It seems to me that this is still the case [here](https://github.com/polylang/polylang-pro/blob/72cc51ad452c46798aa1d7dff934e4496e96dea6/modules/CLI/Command/Language.php#L97-L101) too.

Am I wrong?

Reply:
Sorry, GH made a mess in my comment and my previous one disappeared.
I understand: it means that we will have to modify this code again later to allow unknown locale when WP CLI PR will be finished.